File: core/aop/src/main/java/io/nopecho/distributed/config/RedissonConfig.java
package io.nopecho.distributed.config;

import org.redisson.Redisson;
import org.redisson.api.RedissonClient;
import org.redisson.config.Config;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class RedissonConfig {

    @Value("${distributed-lock.redisson.host}")
    private String host;

    @Value("${distributed-lock.redisson.port}")
    private int port;

    @Bean
    public RedissonClient redissonClient() {
        Config config = new Config();
        config.useSingleServer().setAddress(host + ":" + port);

        return Redisson.create(config);
    }
}

File: core/aop-functional/src/main/kotlin/io/nopecho/distributed/config/RedissonConfig.kt
package io.nopecho.distributed.config

import org.redisson.Redisson
import org.redisson.api.RedissonClient
import org.redisson.config.Config
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.boot.context.properties.ConfigurationPropertiesScan
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@ConfigurationPropertiesScan
@Configuration
class RedissonConfig(
    private val properties: LockProperties,
) {

    @Bean
    @ConditionalOnMissingBean
    fun redissonClient(): RedissonClient {
        val config = Config().apply {
            useSingleServer().setAddress(properties.getSingleRedissonAddress())
        }
        return Redisson.create(config)
    }
}